src/main.rs

```diff
@@ -20,81 +24,103 @@ impl LinkedList {
         Self {
             head: None,
             last: None,
-            current_iter_elem: None,
         }
     }
 
-    fn add_before_head(&mut self, value: i32) {
-        if self.head.is_none() {
+    fn push_front(&mut self, value: i32) {
+        if let Some(head) = &self.head {
             self.head = Some(Rc::new(RefCell::new(Node {
                 data: value,
-                next: None,
+                next: Some(Rc::clone(head)),
             })));
         } else {
-            let copy_of_head = Rc::clone(self.head.as_ref().unwrap());
             self.head = Some(Rc::new(RefCell::new(Node {
                 data: value,
-                next: Some(Rc::clone(&copy_of_head)),
+                next: None,
             })));
         }
-        self.current_iter_elem = self.head.clone();
     }
 
-    fn add_after_n(&mut self, value: i32, mut n: i32) {
-        let mut current = self.head.clone();
+    fn add_after_n(&mut self, value: i32, index: i32) -> Result<(), String> {
+        if self.head.is_none() || self.size() < index {
```

Comment:
```suggestion
        if self.size() < index {
```

По идеи смысл ровно тот же самый